File: Python/Problem009.py
from math import floor,sqrt
from time import time


isTriplet = lambda a, b, c: a < b < c and a**2 + b**2 == c**2 and a+b+c == 1000

def main():
    for i in range(1000):
        for j in range(1000):
            k = (i**2 + j**2)**0.5
            if isTriplet(i, j, int(k)):
                return i*j*int(k)

start = time()
ans = main()

if __name__ == '__main__':
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")

File: Go_Solutions/Solutions/projeuler/Problem009.go
package euler

import (
	"fmt"
	"math"
	"time"
)

/*
Problem009 answers the problem at : https://projecteuler.net/problem=9
	* Problem 9:
		There exists exactly one Pythagorean triplet for which a + b + c = 1000 (a < b < c).
		Find the product abc.
*/
func Problem009() {

	start := time.Now()
	ans := get9()
	end := time.Now()

	fmt.Printf("\nAnswer to Problem 9 : %d\n", ans)
	fmt.Printf("Time Taken: %f seconds\n\n", end.Sub(start).Seconds())
}

func isTriplet(a, b, c int) bool {
	return c*c == a*a+b*b && a < b && b < c
}

func get9() int {
	for i := 1; i < 500; i++ {
		for j := 1; j < 500; j++ {
			k := math.Sqrt(float64(i*i + j*j))

			if isTriplet(i, j, int(k)) && i+j+int(k) == 1000 {
				return i * j * int(k)
			}
		}
	}
	return 0
}
